### PR TITLE
[CELEBORN-668][FOLLOWUP] Handle unknown worker should also reply WorkerLostResponse

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Now worker will send WorkLost too, master should also reply WorkerLostResponse when it's unknown worker

### CODE REVIEW

**Comment by pan3793** [comment]
@waitinfuture any concerns? As we are actively changing code in this area, better to make this one get in first

**Comment by AngersZhuuuu** [comment]
ping @waitinfuture WDYT?

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala [hunk 1]
```diff
@@ -464,13 +464,14 @@ private[celeborn] class Master(
...
         s" for WorkerLost handler!")
-      return
-    }
-
-    statusSystem.handleWorkerLost(host, rpcPort, pushPort, fetchPort, replicatePort, requestId)
-
-    if (context != null) {
-      context.reply(WorkerLostResponse(true))
+      if (context != null) {
+        context.reply(WorkerLostResponse(false))
```
Comment (RexXiong) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:468]: It's ambiguous when reply(WorkerLostResponse with false), how about keep replay true( or false with code-UnKnow).

Comment (AngersZhuuuu) [reply master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala]: Change to true.

File: master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala [hunk 2]
```diff
@@ -464,13 +464,14 @@ private[celeborn] class Master(
...
       logWarning(s"Unknown worker $host:$rpcPort:$pushPort:$fetchPort:$replicatePort" +
         s" for WorkerLost handler!")
-      return
-    }
-
-    statusSystem.handleWorkerLost(host, rpcPort, pushPort, fetchPort, replicatePort, requestId)
-
-    if (context != null) {
-      context.reply(WorkerLostResponse(true))
+      if (context != null) {
```
Comment (RexXiong) [inline master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala:467]: How about following?
```java
    if (worker == null) {
      logWarning(s"Unknown worker $host:$rpcPort:$pushPort:$fetchPort:$replicatePort" +
        s" for WorkerLost handler!")
    } else {
      statusSystem.handleWorkerLost(host, rpcPort, pushPort, fetchPort, replicatePort, requestId)
    }
    
    if (context != null) {
      context.reply(WorkerLostResponse(true))
    }
```

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! Merging to main/0.3

**Pull request closed without merging**